File: lectures/lecture02/geometricshapes_rust/src/square.rs
use crate::geometricshape::GeometricShape;

pub struct Square {
    pub x: i32,
    pub y: i32,
    pub length: i32,
}

impl GeometricShape for Square {
    fn move_to(&mut self, x: i32, y: i32) {
        self.x = x;
        self.y = y;
    }
    
    fn draw(&self) {
        // drawing implementation goes here...
    }

    fn to_string(&self) -> String {
        String::from("square")
    }
}

File: courses/cmpt225/lecture02/geometricshapes_java/src/main/java/example/geometricshapes/Square.java
package example.geometricshapes;

public class Square extends Rectangle {

	public Square(int initX, int initY, int initL) {
		super(initX, initY, initL, initL);
	}
	
	public void setLength(int newL)
	{
		super.setLength(newL);
		setWidth(newL);
	}

	public void draw()
	{
		// drawing implementation...
	}

	@Override
	public String toString() {
		return "Square: top-left=(" + x + "," + y +"), length = " + length;
	}


}


